modules/nux/modules.php

```diff
@@ -260,47 +273,264 @@ public function process() {
     }
 }
 
+/**
+ * @subpackage nux/handler
+ */
+class Hm_Handler_process_import_accouts_servers extends Hm_Handler_Module
+{
+    public function process()
+    {
+        list($success, $form) = $this->process_form(array('accounts_source'));
+
+        if ($success) {
+            $file = $this->request->files['accounts_sample'];
+            if (!is_null($file) && $file['type'] == 'application/x-yaml') {
+                try {
+                    $servers = Yaml::parseFile($file['tmp_name']);
+                } catch (\Throwable $th) {
+                    Hm_Msgs::add('ERR' . $th->getMessage());
+                    $this->save_hm_msgs();
+                    Hm_Dispatch::page_redirect('?page=servers');
+                }
+                foreach ($servers as $key => $value) {
+                    if (empty($value['user']['username']) or empty($value['user']['password'])) {
+                        Hm_Msgs::add('ERRUsername and password are required for: ' . $key . ' server');
+                        return;
+                    }
+                    if ($value['jmap'] && !empty($value['jmap']['jmap_server'])) {
+                        if (!$this->module_is_supported('jmap')) {
+                            Hm_Msgs::add("ERRJMAP module is not enabled");
+                            return;
+                        }
+                        connect_to_imap_server(
```

Comment:
I think we can do it like this:
If IMAP and JMAP are set at the same time, save only JMAP.
Test if JMAP is defined, save it and retrieve its id, otherwise retrieve the IMAP id. If both are not defined, the SMTP is saved.
If IMAP/JMAP is defined and the server failed to register do not also register the corresponding SMTP.
On the other hand, if IMAP/JMAP is registered and the corresponding SMTP fails to be registered, also delete the corresponding IMAP/JMAP.
This way we will ensure that if we register a server all the corresponding information will be recorded and in the event of failure everything will not be saved.